### PR TITLE
hibernate natural id test added

### PR DESCRIPTION
**Pull request by mustafaiman**
Adds a single test for natural id region of hibernate 4

### CODE REVIEW

File: hazelcast-hibernate/hazelcast-hibernate4/src/test/java/com/hazelcast/hibernate/HibernateStatisticsTestSupport.java [hunk 1]
```diff
@@ -205,4 +221,38 @@ public void testUpdateQueryCausesInvalidationOfEntireRegion() {
         assertEquals(10, dummyEntityCacheStats.getMissCount());
         assertEquals(0, dummyEntityCacheStats.getHitCount());
     }
+
+    @Test
+    public void testNaturalIdCacheStillHitsAfterIrrelevantNaturalIdUpdate() {
```
Comment (mesutcelik) [inline hazelcast-hibernate/hazelcast-hibernate4/src/test/java/com/hazelcast/hibernate/HibernateStatisticsTestSupport.java]: I would do the test as below. 
insert(1)
query inserted entity.
assertCacheHit
update inserted entity.
assertMissHit

What do you think? That would be simpler.

File: hazelcast-hibernate/hazelcast-hibernate4/src/test/java/com/hazelcast/hibernate/entity/AnnotatedEntity.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    @GenericGenerator(name="increment", strategy = "increment")
+    public Long getId() {
+        return id;
+    }
+
+    private void setId(Long id) {
+        this.id = id;
+    }
+
+    @NaturalId(mutable = true)
```
Comment (mesutcelik) [inline hazelcast-hibernate/hazelcast-hibernate4/src/test/java/com/hazelcast/hibernate/entity/AnnotatedEntity.java]: can't we put this somewhere on top? At my first check, I just did'nt see `@NaturalId` annotation...

**Comment by mesutcelik** [comment]
:+1: 2 minor comments. Thanks!

I just realized `@NaturalId` has been introduced in hibernate4 so no need to port this to hibernate3

**Comment by mesutcelik** [comment]
@hasancelik ,
have you seen this before?
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-pr-builder/15583/console

```
16:01:35 ERROR: Build step failed with exception
16:01:35 java.lang.NullPointerException: no workspace from node com.cloudbees.jenkins.plugins.mtslavescloud.MansionSlave[2143aba8] which is computer com.cloudbees.jenkins.plugins.mtslavescloud.MansionComputer@5bc4f7d5 and has channel null
```

File: hazelcast-hibernate/hazelcast-hibernate4/src/test/java/com/hazelcast/hibernate/HibernateStatisticsTestSupport.java (hunk 1 above)
Comment (mustafaiman) [inline hazelcast-hibernate/hazelcast-hibernate4/src/test/java/com/hazelcast/hibernate/HibernateStatisticsTestSupport.java]: i split the test into two. one checks if natural id eviction works on updates. the other ensures the eviction is done for only affected entities, not the whole table

**Comment by mustafaiman** [comment]
verify

**Comment by mesutcelik** [comment]
:+1:

**Comment by mustafaiman** [comment]
verify

**Comment by kobalski** [comment]
:+1:

**Pull request merged**